service/src/java/org/apache/hive/service/server/KillQueryImpl.java

```diff
@@ -116,9 +117,21 @@ public static void killChildYarnJobs(Configuration conf, String tag, String doAs
 
   private static boolean isAdmin() {
     boolean isAdmin = false;
-    if (SessionState.get().getAuthorizerV2() != null) {
+    SessionState ss = SessionState.get();
+    if(!HiveConf.getBoolVar(ss.getConf(), HiveConf.ConfVars.HIVE_AUTHORIZATION_ENABLED)) {
+      // If authorization is disabled, hs2 process owner should have kill privileges
       try {
-        SessionState.get().getAuthorizerV2()
+        String currentUser = ss.getUserName();
+        String loginUser = UserGroupInformation.getCurrentUser().getShortUserName();
+        return (currentUser != null) && currentUser.equals(loginUser);
```

Comment:
if loginUser will never be null then use  "return StringUtils.equals(currentUser, loginUser)". this will handle null values also.